### PR TITLE
feat(be): API Test for Datasource

### PR DESCRIPTION
**Pull request by 530051970**
**What is the current behavior?

Please describe the current behavior of the code that you are trying to modify. This could include any issues or limitations that you are addressing.

**What is the updated behavior?

Please describe the changes you are making to the code's behavior. This could include new features or enhancements that you are adding, as well as how they differ from the current behavior.

**Checklist

Please check off the following items before submitting your pull request:

- [x] I have tested my changes.
- [x] No sensitive information is included.
- [x] I've reviewed my changes to ensure they won't cause any new issues that could affect the stability or performance of the codebase.

Please check off the following items if this changes include API modefications:

- [x] I confirm this changes has been test with [authorization validation steps](https://github.com/awslabs/sensitive-data-protection-on-aws/blob/main/CONTRIBUTING.md#contributing-via-pull-requests), it won't break the authorization token verification mechanism.
- [x] I confirm this changes won't cause security issues.

### CODE REVIEW

**Review by yanbasic — APPROVED** [review]
LGTM

**Comment by NingLu** [comment]
We identified new issues on unchanged lines of code. Navigate to the Amazon CodeGuru Reviewer console to view the recommendations to fix them.

File: source/constructs/api/data_source/service.py [hunk 1]
```diff
@@ -408,216 +406,219 @@ def pre_sync(jdbc: SourceJDBCConnection):
...
                 except Exception as e:
-                    logger.info("sync_rds_connection get_crawler and create:")
-                    logger.info(str(e))
-                    response = glue.create_crawler(
+                    sleep(SLEEP_MIN_TIME)
+                    num_retries -= 1
+                else:
+                    break
+            else:
+                raise Exception('UNCONNECTED')
```
Comment (NingLu) [inline source/constructs/api/data_source/service.py:523]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.

File: source/constructs/api/data_source/service.py [hunk 2]
```diff
@@ -2080,3 +2084,52 @@ def query_full_provider_resource_infos():
...
+                Permissions=['ALL'],
+                PermissionsWithGrantOption=['ALL']
+            )
+        except Exception as e:
+            sleep(SLEEP_MIN_TIME)
+            num_retries -= 1
+        else:
+            break
+    else:
+        raise Exception('UNCONNECTED')
```
Comment (NingLu) [inline source/constructs/api/data_source/service.py:2135]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.

**Pull request merged**